react-number-input.jsx

```diff
@@ -82,20 +84,22 @@ var NumberInput = React.createClass({
 	},
 
 	_onChange: function (event) {
-		var value = event.target.value;
+		var target = event.target;
+		var value = target.value;
+		var id = target.id;
 
 		// If current value is all zeroes, let the editing continue but
 		// broadcast onChange event to parent with value of 0.
 		if (value.search(/^0+$/g) === 0) {
 			this.setState(
 				{ value: value },
-				function () { this.props.onChange(0); }
+				function () { this.props.onChange(0, id); }
 			);
 		} else {
 			value = unformat(event.target.value);
 			this.setState(
 				{ value: value },
-				function () { this.props.onChange(value); }
+				function () { this.props.onChange(value, id); }
```

Comment:
> Any reason why we don't just send the event object back?

See http://stackoverflow.com/questions/26317985/react-js-onclick-event-returning-all-null-values

> Also, it may be a good idea to mimic whatever is being returned here in the change event in blur and focus.

Yep, will do that in this PR.